X10D/src/Core/IntrinsicExtensions.cs

```diff
@@ -0,0 +1,103 @@
+﻿#if NETCOREAPP3_0_OR_GREATER
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+using System.Runtime.Intrinsics.Arm;
+using System.Diagnostics.Contracts;
+using System.Runtime.CompilerServices;
+using System.Numerics;
+
+namespace X10D.Core;
+
+/// <summary>
+///     Extension methods for SIMD vectors, namely <see cref="Vector64{T}"/>, <see cref="Vector128{T}"/> and <see cref="Vector256{T}"/>.
+/// </summary>
+public static class IntrinsicExtensions
+{
+    /// <summary>
+    ///     Correcting <see cref="Vector64{T}"/> of <see langword="byte"/> into standard boolean values.
+    /// </summary>
+    /// <param name="vector">Vector of byte to correct.</param>
+    /// <returns>Corrected boolean in form of <see cref="Vector64{T}"/> of bytes.</returns>
+    /// <remarks>This method will ensure that every value can only be 0 or 1. Values of 0 will be kept, and others will be set to 1.</remarks>
+    [Pure]
+    [MethodImpl(MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization)]
+    public static Vector64<byte> CorrectBoolean(this Vector64<byte> vector)
```

Comment:
This method doesn't document PlatformNotSupportedException. Which - currently - will unconditionally be thrown if `AdvSimd.IsSupported` is false.